docs/source/internals.rst

```diff
@@ -101,83 +83,326 @@ Task
+                         exec_pre, rank_start, app_start, app_*, app_stop,
+                         rank_stop, exec_post, exec_stop, launch_collect,
+                         launch_post, launch_stop, task_run_stop
+   * per task          : task_run_cancel_start, task_run_cancel_stop
+
+   NOTE: raptor tasks will not log the complete set of events - they will miss
+   the launch_* events (raptor has not separate launcher), the exec_pre and
+   exec_post events (pre and post exec are not supported), and the task_mkdir_*
+   events (raptor tasks don't have individual sandboxes).
+
+AgentStagingOutputComponent (Component)
+---------------------------------------
+
+::
+
+   staging_stdout_start: reading task stdout starts                 (uid: task)
+   staging_stdout_stop : reading task stdout stops                  (uid: task)
+   staging_stderr_start: reading task stderr starts                 (uid: task)
+   staging_stderr_stop : reading task stderr stops                  (uid: task)
+   staging_uprof_start : reading task profile starts                (uid: task, [APP])
+   staging_uprof_stop  : reading task profile stops                 (uid: task, [APP])
+   staging_out_start   : staging request starts                     (uid: task, msg: did, [Task-DS])
+   staging_out_skip    : staging request is not handled here        (uid: task, msg: did, [Task-DS])
+   staging_out_fail    : staging request failed                     (uid: task, msg: did, [Task-DS])
+   staging_out_stop    : staging request stops                      (uid: task, msg: did, [Task-DS])
+
+   partial orders
+   * per task          : staging_stdout_start, staging_stdout_stop,
+                         staging_stderr_start, staging_stderr_stop,
+                         staging_uprof_start,  staging_uprof_stop,
+   * per file          : staging_out_skip \
+                       | (staging_out_start, (staging_out_fail | staging_out_stop))
+
+UMGRStagingOutputComponent (Component)
+--------------------------------------
+
+::
+
+   staging_out_start   : staging request starts                     (uid: task, msg: did, [Task-DS])
+   staging_out_stop    : staging request stops                      (uid: task, msg: did, [Task-DS])
+
+   partial orders
+   * per file          : staging_out_start, staging_out_stop
+
+UpdateWorker (Component)
+------------------------
+
+::
+
+   update_request      : a state update is requested                (uid: task, msg: state)
+   update_pushed       : bulk state update has been sent            (           msg: 'bulk size: %d')
+   update_pushed       : a state update has been send               (uid: task, msg: state)
+
+   partial orders
+   * per state update  : update_request, update_pushed
+
+All profiles
+------------
+
+::
+
+   sync_abs            : sets an absolute, NTP synced time stamp               ([INTERNAL])
+   sync_rel            : sets a *pair* of time stamps considered simultaneous  ([INTERNAL])
```

Comment:
`sync_rel` looks like it is obsolete everywhere (some other components above also referred it) and we only use `sync_abs`

Reply:
I removed all the `sync_rel`